### PR TITLE
validate maximum length of account number and bank blocks

### PR DESCRIPTION
**Pull request by JonSchapiro**

### CODE REVIEW

**Comment by shahraizali** [comment]
Hey @JonSchapiro  Are these test cases working for you locally?

**Comment by JonSchapiro** [comment]
> Hey @JonSchapiro Are these test cases working for you locally?

@shahraizali my second test now works. However, it seems that the validator is not working on the query param while asking for an account by account_number?

**Comment by shahraizali** [comment]
okay, btw if don't know already. you can run the tests locally as well so you don't have to push to test. I am judging from multiple commits.

**Comment by JonSchapiro** [comment]
> okay, btw if don't know already. you can run the tests locally as well so you don't have to push to test. I am judging from multiple commits.

yes, apologies. I realized that now and can run them locally.

**Comment by shahraizali** [comment]
> yes, apologies.

No worries. have at it, try to debug the issue now that you can run it locally. make sure you are pinging the correct endpoint and also make sure you are sending the account number along in the expected and correct way. You can update here or in the channel with your findings.

**Comment by JonSchapiro** [comment]
https://github.com/thenewboston-developers/Validator/issues/124

**Comment by JonSchapiro** [comment]
@shahraizali could I have a review on this please :)?

**Review by shahraizali — APPROVED** [review]

**Review by vosi — APPROVED** [review]

**Comment by JonSchapiro** [comment]
@vosi @shahraizali can this be merged?

**Comment by JonSchapiro** [comment]
@shahraizali @vosi just checking in again to see if someone can merge?

File: v1/accounts/factories/account.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 
 class AccountFactory(DjangoModelFactory):
-    account_number = Faker('pystr', max_chars=VERIFY_KEY_LENGTH)
+    account_number = Faker('pystr', max_chars=VERIFY_KEY_LENGTH - 1)
```
Comment (vosi) [inline v1/accounts/factories/account.py:9]: why -1?

File: v1/accounts/views/account.py [hunk 2]
```diff
@@ -31,12 +33,18 @@ class AccountViewSet(
 
     @action(methods=['get'], detail=True)
     def balance(self, request, account_number=None):
+        if account_number is not None and len(account_number) >= VERIFY_KEY_LENGTH:
+            return Response({'balance': None}, status=status.HTTP_401_UNAUTHORIZED)
```
Comment (vosi) [inline v1/accounts/views/account.py:37]: why 401?

File: v1/accounts/views/account.py [hunk 3]
```diff
@@ -31,12 +33,18 @@ class AccountViewSet(
 
     @action(methods=['get'], detail=True)
     def balance(self, request, account_number=None):
+        if account_number is not None and len(account_number) >= VERIFY_KEY_LENGTH:
```
Comment (vosi) [inline v1/accounts/views/account.py:36]: >= ? 64 is ok if I'm not mistaking.

**Review by vosi — CHANGES_REQUESTED** [review]

**Comment by buckyroberts** [comment]
Closing due to inactivity. Want to allow another contributors to work on these tests.

**Pull request closed without merging**